pkg/cmd/pomerium/pomerium.go

```diff
@@ -30,9 +31,30 @@ import (
 	"github.com/pomerium/pomerium/proxy"
 )
 
+type RunOptions struct {
+	fileMgr *filemgr.Manager
+}
+
+type RunOption func(*RunOptions)
+
+func (o *RunOptions) apply(opts ...RunOption) {
+	for _, op := range opts {
+		op(o)
+	}
+}
+
+func WithOverrideFileManager(fileMgr *filemgr.Manager) RunOption {
```

Comment:
The `fileMgr` changes aren't directly related to contexts and loggers, are they? Would you mind updating the PR description to include the rationale for these changes (or maybe split these off into a separate PR)?